experimental/framework/fluid-static/src/containerCode.ts

```diff
@@ -46,6 +47,57 @@ export interface FluidContainer
      * The returned object needs to be stored via handle by the caller.
      */
     create<T extends IFluidLoadable>(objectClass: LoadableObjectClass<T>): Promise<T>;
+
+    /**
+     * If the container is created in a detached state, this will connect it to the service. This allows the
+     * data in the container to be permanently persisted instead of being in-memory. This will throw
+     * an error if the container is already attached when this is called.
+     */
+    attachToService: (config: C) => Promise<void>;
+}
+
+export class FluidContainer<C> extends EventEmitter implements IFluidContainer<C> {
+    private readonly connectedHandler = (id: string) => this.emit("connected", id);
+
+    constructor(
+        private readonly rootDataObject: RootDataObject,
+        private readonly container: Container,
+        private readonly attachToServiceCallback: (config: C, container: Container) => Promise<void>)
+    {
+        super();
+        this.rootDataObject.on("connected", this.connectedHandler);
+    }
+
+    public get initialObjects(): LoadableObjectRecord {
+        return this.rootDataObject.initialObjects;
+    }
+
+    public get audience(): IAudience {
+        return this.rootDataObject.audience;
+    }
+
+    public get clientId() {
+        return this.rootDataObject.clientId;
+    }
+
+    public get attachState(): AttachState {
+        return this.rootDataObject.attachState;
+    }
+
+    public async create<T extends IFluidLoadable>(
+        objectClass: LoadableObjectClass<T>,
+    ): Promise<T> {
+        return this.rootDataObject.create<T>(objectClass);
+    }
+
+    public async attachToService(config: C): Promise<void> {
+        if (this.attachState === AttachState.Detached) {
+            await this.attachToServiceCallback(config, this.container);
+        } else {
+            const errorString = this.attachState === AttachState.Attached ? "already attached" : "currently attaching";
+            throw Error(`Container is ${errorString}`);
```

Comment:
is this throw going to cause problems with live reload? should it just console log the error and skip attaching?

Reply:
This shouldn't affect live reload as it is an explicit call that the user makes the container's API. It shouldn't get re-triggered on each load of the app

Reply:
attach is idempotent, why make it not

Reply:
Made it a warning so the user is aware that they are attempting to re-attach, but it does not block them with an error